### PR TITLE
Types: split single index.ts file into files based on function/category

### PR DESCRIPTION
**Pull request by filmaj**
###  Summary

This is purely a text-movement PR; no changes (other than a few comments at the top of certain files) from the base.

I'm mainly looking for feedback on the organization of the files. If you have feedback on directory structure or file names, let me know!

### CODE REVIEW

File: packages/types/src/block-kit/blocks.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+export interface Block {
+  type: string;
+  block_id?: string;
+}
+
+export type KnownBlock = ImageBlock | ContextBlock | ActionsBlock | DividerBlock |
+SectionBlock | InputBlock | FileBlock | HeaderBlock | VideoBlock;
+
+export interface ActionsBlock extends Block {
```
Comment (srajiang) [inline packages/types/src/block-kit/blocks.ts:19]: Seems worth it to adding additional annotation to more easily find the documentation link while we're at it? 

Fine to punt to the later TODO moment referenced in the comments above as well. 

```suggestion
// Reference: https://api.slack.com/reference/block-kit/blocks#actions
export interface ActionsBlock extends Block {
```

File: packages/types/src/block-kit/elements.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
+// This file contains objects documented here: https://api.slack.com/reference/block-kit/block-elements
```
Comment (srajiang) [inline packages/types/src/block-kit/elements.ts:1]: Should this just be called block-elements.ts as well based on the documentation? Would line up better alphabetically with blocks.ts and makes it clear they belong to blocks

File: packages/types/src/dialog.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/**
```
Comment (srajiang) [inline packages/types/src/dialog.ts:1]: Nice that we'll just be able to poof this file when major version bump comes along 🧹 !

File: packages/types/src/views.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { Block, KnownBlock } from './block-kit/blocks';
+import { PlainTextElement } from './block-kit/composition-objects';
+
+export interface HomeView {
```
Comment (srajiang) [inline packages/types/src/views.ts:4]:
```suggestion
// Reference: https://api.slack.com/surfaces/app-home#composing
export interface HomeView {
```

File: packages/types/src/views.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  type: 'home';
+  blocks: (KnownBlock | Block)[];
+  private_metadata?: string;
+  callback_id?: string;
+  external_id?: string;
+}
+
+export interface ModalView {
```
Comment (srajiang) [inline packages/types/src/views.ts:12]:
```suggestion
// Reference: https://api.slack.com/surfaces/modals#composing_views
export interface ModalView {
```

**Review by srajiang — COMMENTED** [review]
I like the structure! Left a few small suggestions most notably renaming elements.ts as block-elements.ts.

File: packages/types/src/block-kit/blocks.ts (hunk 1 above)
Comment (filmaj) [reply packages/types/src/block-kit/blocks.ts:19]: Yes exactly, I will be doing exactly these kind of changes in the next pull request 😄 I didn't want to add more additional lines in this specific PR as the diff would be too confusing given the movement of code across files. I think that in the next PR, though, these kinds of documentation additions you suggest would be clearer and easier to review.

File: packages/types/src/block-kit/elements.ts (hunk 2 above)
Comment (filmaj) [reply packages/types/src/block-kit/elements.ts:1]: Yes excellent point! Will do

**Review by srajiang — APPROVED** [review]
Great, LGTM!

**Pull request merged**